lib/X25519KeyPair.js

```diff
@@ -0,0 +1,321 @@
+/*!
+ * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const forge = require('node-forge');
+const {util: {binary: {base58}}} = forge;
+const {LDKeyPair} = require('crypto-ld');
+const {base58Decode} = require('crypto-ld/lib/util');
+const ed2curve = require('ed2curve');
+const nacl = require('tweetnacl');
+
+class X25519KeyPair extends LDKeyPair {
+  /**
+   * An implementation of x25519
+   * [X25519 Key Agreement 2019]{@link https://w3c-dvcg.github.io/} representation
+   * @example
+   * > const privateKeyBase58 =
+   *   '...';
+   * > const options = {
+   *   publicKeyBase58: '...',
+   *   privateKeyBase58
+   * };
+   * > const DHKey = new X25519KeyPair(options);
+   * > DHKey
+   * X25519KeyPair { ...
+   * @param {KeyPairOptions} options - Base58 keys plus other options
+   * @param {string} options.controller
+   * @param {string} options.id
+   * @param {string} options.publicKeyBase58 - Base58 encoded Public Key
+   * unencoded is 32-bytes.
+   * @param {string} options.privateKeyBase58 - Base58 Private Key
+   * unencoded is 32-bytes.
+   */
+  constructor(options = {}) {
+    super(options);
+    this.type = 'X25519KeyAgreementKey2019';
+    this.privateKeyBase58 = options.privateKeyBase58;
+    this.publicKeyBase58 = options.publicKeyBase58;
+  }
+  /**
+   * Returns the Base58 encoded public key.
+   * @implements {LDKeyPair#publicKey}
+   * @readonly
+   *
+   * @returns {string} The Base58 encoded public key.
+   * @see [publicKey]{@link ./LDKeyPair.md#publicKey}
```

Comment:
Should these links, here and below, to crypto-ld docs be specified some other way?

Reply:
I'm honestly not sure, I didn't put them in there. I think it's advanced JSDoc usage?

Reply:
I assumed they were copied from crypto-ld code that does have that md file.

Reply:
Ah, ok. Will remove.